Assignment6/MyArray.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Assignment6 {
+    public class MyArray<Data> : ICollection<Data> {
```

Comment:
```suggestion
    public class MyArray<TCollection> : ICollection<TCollection> {
```
You should be getting a warning for this as the generic type is usually prefixed with a T. Calling your generic Data may be confusing.

Reply:
No error, but I'll fix that. Thanks!